### ISSUE TITLE
Show module contents (C-c C-o) prints garbled names in clause

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#3687]
Show module contents (`C-c C-o`) shuffles bound variables badly in its output.
```agda
open import Agda.Primitive
open import Agda.Builtin.Equality

-- Setoid

record Setoid c ℓ : Set (lsuc (c ⊔ ℓ)) where
  infix 4 _≈_
  field
    Carrier       : Set c
    _≈_           : Carrier → Carrier → Set ℓ

-- Category

record Category o h e : Set (lsuc (o ⊔ h ⊔ e)) where
  field
    Obj  : Set o
    HomS : (A B : Obj) → Setoid h e

  open module HomSetoid  (A B : Obj) = Setoid (HomS A B)
    public using ()
    renaming ( Carrier       to Hom )

  open module HomSetoidH {A B : Obj} = HomSetoid A B
    public using ()
    renaming ( _≈_           to Eq
             )
  field
    id    : (A : Obj) → Hom A A
    comp  : ∀{A B C} (f : Hom B C) (g : Hom A B) → Hom A C

open Category public

-- Functor

record Functor {o1 h1 e1} (C1 : Category o1 h1 e1)
               {o2 h2 e2} (C2 : Category o2 h2 e2)
       : Set (o1 ⊔ h1 ⊔ e1 ⊔ o2 ⊔ h2 ⊔ e2) where
  field
    App : C1 .Obj → C2 .Obj
    map : ∀{A B : C1 .Obj}
        → C1 .Hom A B
        → C2 .Hom (App A) (App B)

open Functor public

-- Functor composition

compFun : ∀ {o1 o2 o3 e1 e2 e3 h1 h2 h3}
            {C1 : Category o1 e1 h1}
            {C2 : Category o2 e2 h2}
            {C3 : Category o3 e3 h3}
        → (F : Functor C2 C3) (G : Functor C1 C2) → Functor C1 C3
compFun F G .App A = F .App (G .App A)
compFun F G .map f = F .map (G .map f)

-- Functor equality

module _ {o e h} (C D : Category o e h) where

  record FunEq (F G : Functor C D) : Set (lsuc lzero ⊔ o ⊔ h ⊔ e) where
    field
      eqApp : ∀{c} → F .App c ≡ G .App c
      eqMap : ∀{c d} (f g : C .Hom c d) (eq : C .Eq f g) → Set
  open FunEq

  fun-comp-assoc : ∀ {A B : Category o e h}
                   (F : Functor B D) (G : Functor A B) (H : Functor C A) →
                 FunEq (compFun (compFun F G) H) (compFun F (compFun G H))
  fun-comp-assoc F G H .eqApp = refl
  fun-comp-assoc F G H .eqMap f g f=g = {!D!}  -- C-c C-o

-- Names
--   Obj  : Set o
--   HomS : f=g → f=g → Setoid o e
--   id   : (A : g) → f=g A A .Setoid.Carrier
--   comp : {A B : f} {C = C₁ : f} →
--          g B C₁ .Setoid.Carrier →
--          g A B .Setoid.Carrier → g A C₁ .Setoid.Carrier
```
In the output, the names are completely garbled.

I'll do some further shrinking.

**Comment by andreasabel** [comment]
Same amount of garbage for less code.
```agda
record Cat : Set₁ where
  field
    Obj   : Set
    Hom   : (A B : Obj) → Set
    Eq    : ∀{A B} (f g : Hom A B) → Set
    id    : (A : Obj) → Hom A A
    comp  : ∀{A B C} (f : Hom B C) (g : Hom A B) → Hom A C

record Functor (C1 C2 : Cat) : Set where

record FunEq {C D : Cat} (F G : Functor C D) : Set₁ where
  field
    eqMap : ∀{c d} (f g : C .Cat.Hom c d) (eq : C .Cat.Eq f g) → Set

test : ∀ C D (F G : Functor C D) → FunEq F G
test C D F G .FunEq.eqMap f g f=g = {!D!}  -- C-c C-o

-- Names
--   Obj  : Set
--   Hom  : f=g → f=g → Set
--   Eq   : {A B : g} → f=g A B → f=g A B → Set
--   id   : (A : f) → g A A
--   comp : {A B : d} {C = C₁ : d} → f B C₁ → f A B → f A C₁
```
Seems to be a bug in feature #1926 which exists since 2.5.1.1.

**Comment by andreasabel** [comment]
MWE:
```agda
open import Agda.Builtin.Sigma

test : {A : Set} {B : A → Set} (r : Σ A B) → Set
test r = {!r!} -- C-c C-o
```
```
Names
  fst : A
  snd : A r
```

**Comment by asr** [comment]
Why milestone 2.6.0? I'm preparing the release based on  ba50d355f3407c64d948d60042b3e0a31b2b18c6?

**Comment by andreasabel** [comment]
Ok, you can leave this one for 2.6.1, it is not so important. I was not aware you were packing up now.  But see also the new regression #3689.

**Comment by asr** [comment]
Since I need to include the fix of #3689, I'll also include the fix of this issue.
